include/libtorrent/span.hpp

```diff
@@ -105,12 +105,19 @@ namespace libtorrent
 			return { data() + offset, count };
 		}
 
-		T& operator[](std::size_t const idx)
+		T& operator[](std::size_t const idx) const
 		{
 			TORRENT_ASSERT(idx < m_len);
 			return m_ptr[idx];
 		}
 
+		T& operator[](int const idx) const
```

Comment:
I didn't review completely the PR, but I see `span<T>` like a very interoperable type, similar to `std::vector<T>`, and that means classic `std::size_t` index. Maybe you can use `typed_span<T, I>` in the place you need to index by `int`, similar to `aux::vector`.